File: app/src/main/java/org/tvheadend/tvhclient/data/service/worker/EpgDataUpdateWorker.java
package org.tvheadend.tvhclient.data.service.worker;

import android.content.Context;
import android.content.Intent;
import android.support.annotation.NonNull;

import org.tvheadend.tvhclient.data.service.EpgSyncIntentService;

import androidx.work.Worker;
import androidx.work.WorkerParameters;
import timber.log.Timber;

public class EpgDataUpdateWorker extends Worker {

    public EpgDataUpdateWorker(@NonNull Context context, @NonNull WorkerParameters workerParams) {
        super(context, workerParams);
    }

    @NonNull
    @Override
    public Result doWork() {
        Timber.d("Loading more event data from server");

        Intent intent = new Intent();
        intent.setAction("getMoreEvents");
        intent.putExtra("numFollowing", 250);
        EpgSyncIntentService.enqueueWork(getApplicationContext(), intent);
        return Result.success();
    }
}

File: app/src/main/java/org/tvheadend/tvhclient/util/worker/EpgDataUpdateWorker.kt
package org.tvheadend.tvhclient.util.worker

import android.content.Context
import android.content.Intent
import androidx.work.Worker
import androidx.work.WorkerParameters
import org.tvheadend.tvhclient.service.ConnectionIntentService
import timber.log.Timber

class EpgDataUpdateWorker(context: Context, workerParams: WorkerParameters) : Worker(context, workerParams) {

    companion object {
        const val WORK_NAME = "EpgDataUpdateWorker"
    }

    override fun doWork(): Result {
        Timber.d("Loading more event data from server")

        val intent = Intent()
        intent.action = "getMoreEvents"
        intent.putExtra("numFollowing", 250)
        ConnectionIntentService.enqueueWork(applicationContext, intent)
        return Result.success()
    }
}
